makeSpectral.py

```diff
@@ -0,0 +1,257 @@
+from os import listdir, rename
+from os.path import isfile, join, abspath, basename, splitext, dirname
+from typing import override
+from typing import TextIO, List
+from shutil import copy, SameFileError
+from argparse import ArgumentParser
+
+"""
+NOTE:
+	Although a little more cumbersome, I feel that providing the our
+	*replace substring* function extra verbosity is important.
+	This is because the User's chosen Name and Description
+	may occur in parts of the c++ code we do not want to edit.
+	The inclusion of the context around the needle we want to substitute
+	will help prevent accidental substitions.
+Happy Coding. Pix ❤️
+""" 
+
+class Cmd:
+	def __init__(self, p : str, n : str, d : str, v : bool):
+		self.path = p.pop() if p else None
+		self.name = n.pop() if n else None
+		self.description = d.pop() if d else None
+		self.verbose = v.pop() if v else False
+
+class Abstract_JucerFile:
+
+	def __init__(self, name:str, desc:str, path:str):
+		self.name = name
+		self.desc = desc
+		self.path = path
+		self.diry = dirname(path)
+
+
+	def swit(self, haystack:str, needle:str, spore:str) -> str :
+		if haystack == "" or haystack is None:
+			return None
+		if not needle in haystack:
+			raise ValueError("needle not found in haystack")
+		return haystack.replace(needle, spore, 1)
+
+	def manip(self, dt : str) -> str:
+		return dt
+
+	def rename_path(self, old_path : str) -> str:	
+		new_file_name = self.swit(basename(old_path), "Base", self.name)
+		new_file_path = join(self.diry, new_file_name)
+		rename(old_path, new_file_path)
+		return new_file_path
+
+	def crank(self) -> str:
+		print(self.path)
+		with open(self.path, 'r+') as f:
+			text = f.read()
+			manip_text = self.manip(text)
+			f.seek(0)
+			f.write(manip_text)
+			f.truncate()
+		self.path = self.rename_path(self.path)
+		return text
+
+class Base(Abstract_JucerFile):
+
+	def manip_description(self, dt:str) -> str:
+		dt = self.swit(dt, 'pluginDesc="BaseDescription"', f'pluginDesc="{self.desc}"')
+		dt = self.swit(dt, 'name="BaseTitle"', f'name="{self.name}"')
+		return dt
+
+	@override
+	def manip(self, dt : str) -> str:
+		if self.desc:
```

Comment:
So, removing the `BaseTitle` change from `manip_description` now means we should do it here.
I'm running it twice because there's 2 instances. Looks a bit weird, I guess we could add an argument to `swit` but this should be the only time we need to run it twice.

```suggestion
    dt = self.swit(dt, 'name="BaseTitle"', f'name="{self.name}"')
    dt = self.swit(dt, 'name="BaseTitle"', f'name="{self.name}"')
		if self.desc:
```